### DISCUSSION TITLE
After updating to Streamlink 3.1.0, Streamlink works (CLI) but on 'Streamlink Twitch GUI' I get an error launching any stream: Exit code 3221225477

### DISCUSSION THREAD

**Discussion by KaXaSA** [streamlink/streamlink-twitch-gui#854] [Q&A] [answered]
So I can use Streamlink on PowerShell like:
`streamlink https://www.twitch.tv/lirik best --player "C:\Program Files\mpv.net\mpvnet.exe"`

but when I open Streamlink Twitch GUI and try to launch any stream I get this error:
`Exit code 3221225477`

It's weird because Streamlink Twitch GUI was working fine before the Streamlink update, but I tried to ~~downgrade Streamlink~~¹/ uninstall GUI/ remove Streamlink AppData folders, reboot the pc, but even with the default settings (except for the necessary stuff like streamlink executable: C:\Program Files (x86)\Streamlink\bin\streamlinkw.exe), in streamlink GUI I still get the Exit code 3221225477 when launching any stream.

**¹ uninstalling Streamlink and then installing 3.0.1, 3.0.2 OR 3.0.3 'fixed' the issue, so I assume there is a problem with 3.1.0**

Opening
...AppData\Local\Temp\streamlink-twitch-gui\logs
I see a bunch of these:
```
[2022-01-23T16:28:00.555Z][error][StreamingService]
Error: Exit code 3221225477
```

Windows 10 PRO 64 bits - 21H1 - 19043.1466
Streamlink Twitch GUI: v1.13.0
Streamlink: 3.1.0

**Comment by bastimeyer** [comment]
**How did you install Streamlink and Streamlink Twitch GUI?**
Are you using anything special to install it on Windows, like a third party package manager for example?
I'm asking because of https://github.com/streamlink/streamlink-twitch-gui/issues/851#issuecomment-1010567480, as an example of third party programs altering the application contents.

**Have you mixed multiple Streamlink installs on your system?**
Make sure to completely remove Streamlink to prevent any kind of side-effect issues from broken second installs, etc. This is what you'll have to figure out on your own, since I don't know your system. Be aware that Streamlink's installer for Windows allows you to install it in two different ways, per-user and system-wide.

**What is your Streamlink configuration in Streamlink Twitch GUI?**
Since you've already found the error log, please post the entire log output when trying to launch a stream and make sure to remove personal data if there is any (don't remove anything else). Streamlink Twitch GUI first needs to resolve the paths according to your configuration before it can run the executables. All of this will be logged if you launch Streamlink Twitch GUI with the `--loglevel debug` parameter. Please do that. Either do this via the command line shell or add this to the launch parameters in a desktop shortcut, etc, in case you're unfamiliar with this.

**Comment by KaXaSA** [comment]
I used the installer for both, I'm not using choco/ other package managers.

I only have streamlink 3.1.0 installed, no other versions.

Log
```
[2022-01-23T18:53:19.841Z][debug][StreamingService]
Preparing to launch stream
{
    "id": "sodapoppin",
    "quality": "source",
    "low_latency": false,
    "disable_ads": true,
    "chat_open": true,
    "started": "2022-01-23T18:53:19.818Z",
    "stream": "26301881",
    "channel": "26301881"
}

[2022-01-23T18:53:19.844Z][debug][StreamingService]
Resolving streaming provider
{
    "provider": "streamlink",
    "providerUserData": {
        "exec": "C:\\Program Files (x86)\\Streamlink\\bin\\streamlinkw.exe",
        "params": null,
        "pythonscript": null
    }
}

[2022-01-23T18:53:19.852Z][debug][StreamingService]
Found streaming provider
{
    "exec": "C:\\Program Files (x86)\\Streamlink\\bin\\streamlinkw.exe",
    "params": null,
    "env": null
}

[2022-01-23T18:53:19.859Z][debug][StreamingService]
Spawning process
{
    "exec": "C:\\Program Files (x86)\\Streamlink\\bin\\streamlinkw.exe",
    "params": [
        "--version"
    ],
    "env": null
}

[2022-01-23T18:53:19.918Z][error][StreamingService]
Error: Exit code 3221225477
{
    "id": "sodapoppin",
    "quality": "source",
    "low_latency": false,
    "disable_ads": true,
    "chat_open": true,
    "started": "2022-01-23T18:53:19.818Z",
    "stream": "26301881",
    "channel": "26301881"
}
```

(thanks)

**Reply by bastimeyer** [reply]
> ```
> [2022-01-23T18:53:19.859Z][debug][StreamingService]
> Spawning process
> {
>     "exec": "C:\\Program Files (x86)\\Streamlink\\bin\\streamlinkw.exe",
>     "params": [
>         "--version"
>     ],
>     "env": null
> }
> 
> [2022-01-23T18:53:19.918Z][error][StreamingService]
> Error: Exit code 3221225477
> ```

As you can see, it performs the version check via `--version`, to make sure that you're using a Streamlink version that's compatible, but the process is exiting with error code `3221225477`.

This is what should be logged, if Streamlink returns the string `streamlink 3.1.0`:
```
Validated streaming provider
{
    "version": "3.1.0"
}
```

This means that running
```
C:\Program Files (x86)\Streamlink\bin\streamlinkw.exe --version
```
results in exit code 3221225477 instead of exit code 0, so Streamlink Twitch GUI can't validate it.

Please run the command I've just posted and check if there's additional output which the Twitch GUI can't interpret.

Other than that, I can't tell you why it's failing on your system. Streamlink 3.1.0 is a regular minor upgrade compared to 3.0.3 and nothing in regards to the Windows installer or the bundled Python install has been changed. The Windows installer installs Python 3.9.8 + Streamlink + Streamlink dependencies:
https://github.com/streamlink/streamlink/blame/3.1.0/script/makeinstaller.sh#L37

If you google for exit code 3221225477 on Windows (0xC0000005 in hex), you'll see that
> Code 3221225477
> Indicates that the executed program has terminated abnormally or crashed.
> Application Error 0xc0000005 (Access Violation) error is usually caused by your computer not being able to correctly process the files and settings required to run a particular program or installation.

https://answers.microsoft.com/en-us/windows/forum/all/application-error-0xc0000005/6224ae45-a251-4f21-b076-74524618d00a

I don't use Windows, so this is beyond what I can do to help you. You'll have to figure this out on your own, unfortunately. Google the error code in combination with the keyword "Windows" and go from there.

**Reply by KaXaSA** [reply]
> Please run the command I've just posted and check if there's additional output which the Twitch GUI can't interpret.

![image](https://user-images.githubusercontent.com/6237603/150720868-7e220ac5-c311-426a-93ef-3fadacf1eb64.png)

streamlink**w**.exe --version 

doesn't give me any output

Thanks.

**Reply by bastimeyer** [reply]
There seems to be an issue with the `streamlinkw.exe` executable. I going to open a new issue on the streamlink issue tracker and see what needs to be done in order to fix this.

Both `streamlink.exe` and `streamlinkw.exe` are wrappers for Streamlink's Python entry module and the specific Python executable.

`python.exe` always launches a command line window on Windows when it is invoked from a non-command-line context and its output streams then can't be read by other programs. `pythonw.exe` on the other hand doesn't do that, which is why Streamlink Twitch GUI requires `streamlinkw.exe` as the Streamlink executable, so it can read its output.

The issue seems to be with the `pythonw.exe` executable, which is surprising though, as the Python version hasn't been changed.

**Reply by bastimeyer** [reply]
https://github.com/streamlink/streamlink/issues/4307

**Reply by KaXaSA** [reply]
So, I tried to uninstall Streamlink, remove anything I could find related to it (again), then I installed Streamlink 3.0.1, 3.0.2, 3.0.3 and 3.1.0 to test each one of them.

**Versions 3.0.x**  worked fine with **Streamlink Twitch GUI** no errors when launching streams:

![image](https://user-images.githubusercontent.com/6237603/150811022-0d7441c2-7d28-47de-b279-8977905947a7.png)
(the result is the same in all versions, but the log -copied below- from versions 3.0.x now have those extra lines with 'Validated streaming provider')

**Version 3.1.0** error is still the same (3221225477).
 
(The first time I tried to downgrade it didn't work, maybe because I didn't uninstall/remove the _program files_ subfolder first, and tried to install the older versions over it?)

**[Test] Works with Streamlink Twitch GUI?**
3.0.1 ✅¹
3.0.2 ✅¹
3.0.3 ✅¹
3.1.0 ❌²

¹ They work with **Streamlink Twitch GUI**, but the log now have the 'Validated streaming provider' informing Streamlink's version.
² I tried to download again just to make sure it wasn't my fault, but the problem still persists.

Log from 3.0.3:

```
[2022-01-24T15:47:08.122Z][debug][StreamingService]
Preparing to launch stream
{
    "id": "retroislandgaming",
    "quality": "source",
    "low_latency": true,
    "disable_ads": true,
    "chat_open": true,
    "started": "2022-01-24T15:47:08.100Z",
    "stream": "100449129",
    "channel": "100449129"
}

[2022-01-24T15:47:08.127Z][debug][StreamingService]
Resolving streaming provider
{
    "provider": "streamlink",
    "providerUserData": {
        "exec": "C:\\Program Files (x86)\\Streamlink\\bin\\streamlinkw.exe",
        "params": null,
        "pythonscript": null
    }
}

[2022-01-24T15:47:08.128Z][debug][StreamingService]
Found streaming provider
{
    "exec": "C:\\Program Files (x86)\\Streamlink\\bin\\streamlinkw.exe",
    "params": null,
    "env": null
}

[2022-01-24T15:47:08.135Z][debug][StreamingService]
Spawning process
{
    "exec": "C:\\Program Files (x86)\\Streamlink\\bin\\streamlinkw.exe",
    "params": [
        "--version"
    ],
    "env": null
}

[2022-01-24T15:47:08.667Z][debug][StreamingService]
Validated streaming provider
{
    "version": "3.0.3"
}

[2022-01-24T15:47:08.693Z][debug][StreamingService]
Spawning process
{
    "exec": "C:\\Program Files (x86)\\Streamlink\\bin\\streamlinkw.exe",
    "params": [
        "--twitch-disable-hosting",
        "--twitch-disable-ads",
        "--twitch-low-latency",
        "--player",
        "C:\\Program Files\\mpv.net\\mpvnet.exe",
        "--player-args",
        "--profile=twitch1080",
        "--hls-live-edge",
        "2",
        "--hls-segment-threads",
        "10",
        "--retry-open",
        "10",
        "--retry-streams",
        "5",
        "twitch.tv/retroislandgaming",
        "best"
    ],
    "env": null
}
```

**Comment by bastimeyer** [comment, marked as answer]
https://github.com/streamlink/streamlink/releases/tag/3.1.1
